### PR TITLE
[feat] Add fundingRouteFullAmount flag to SM

### PR DESCRIPTION
**Pull request by mimi-imtbl**
### Hi👋, please prefix this PR's title with:

- [ ] `breaking-change:` if you have introduced modification that necessitates immediate adjustments by this SDK's users to their applications, clients, or integrations to avert disruptions to existing features or functionalities.
- [x] `feat:`, `fix:`, `refactor:`, `docs:`, or `chore:`.

# Summary

Add `fundingRouteFullAmount` flag to override fractional balance check.

### CODE REVIEW

**Review by jhesgodi — DISMISSED** [review]

File: packages/checkout/sdk/src/types/smartCheckout.ts [hunk 1]
```diff
@@ -345,6 +345,8 @@ export interface SmartCheckoutParams {
   onComplete?: (result: SmartCheckoutResult) => void;
   /** The callback to be executed on each funding route found */
   onFundingRoute?: (fundingRoute: FundingRoute) => void;
+  /** A flag to get funding routes based pm full or partial amonut */
```
Comment (dreamoftrees) [inline packages/checkout/sdk/src/types/smartCheckout.ts:348]: what does `pm` mean here? also small nit on spelling fix for `amonut`

File: packages/checkout/sdk/src/smartCheckout/smartCheckout.ts [hunk 2]
```diff
@@ -92,6 +93,7 @@ export const smartCheckout = async (
   routingOptions?: AvailableRoutingOptions,
   onComplete?: (result: SmartCheckoutResult) => void,
   onFundingRoute?: (fundingRoute: FundingRoute) => void,
+  fundingRouteFullAmount: boolean = true,
```
Comment (dreamoftrees) [inline packages/checkout/sdk/src/smartCheckout/smartCheckout.ts:96]: In order to be backwards compatible with current behaviour do we need to default this to false? So implementation does fractional by default? Or am I getting the meaning mixed up hehe

**Review by dreamoftrees — COMMENTED** [review]
Nice work @mimi-imtbl just a couple of small items.

File: packages/checkout/sdk/src/types/smartCheckout.ts (hunk 1 above)
Comment (mimi-imtbl) [reply packages/checkout/sdk/src/types/smartCheckout.ts:348]: a my lil cooked brain, good catch Charlie!

File: packages/checkout/sdk/src/smartCheckout/smartCheckout.ts (hunk 2 above)
Comment (mimi-imtbl) [reply packages/checkout/sdk/src/smartCheckout/smartCheckout.ts:96]: good catch, thanks Charlie

**Review by jhesgodi — APPROVED** [review]

**Review by dreamoftrees — APPROVED** [review]
👍

**Pull request merged**